src/renderer/atlas/AtlasEngine.h

```diff
@@ -117,6 +119,17 @@ namespace Microsoft::Console::Render::Atlas
         std::unique_ptr<IBackend> _b;
         RenderingPayload _p;
 
+        // _p.s->font->customGlyphs is the setting which decides whether we should map box drawing glyphs to
+        // our own builtin versions. There's just one problem: BackendD2D doesn't have this functionality.
+        // But since AtlasEngine shapes the text before it's handed to the backends, it would need to know
+        // whether BackendD2D is in use, before BackendD2D even exists. These two flags solve the issue
+        // by triggering a complete, immediate redraw whenever the backend type changes.
+        //
+        // The proper solution is to move text shaping into the backends.
+        // Someone just needs to write a generic "TextBuffer to DWRITE_GLYPH_RUN" function.
+        bool _hackWantsCustomGlyphs = true;
+        bool _hackTriggerRedrawAll = false;

```

Comment:
sorry for yor beautiful architecture

Reply:
![image](https://github.com/microsoft/terminal/assets/2256941/037a426e-2bb6-41ed-abf3-6910fc5afcec)

Reply:
DYING